### ISSUE TITLE
Merge request

### ISSUE THREAD

**Issue by demarcush** [Chocolate4U/Iran-v2ray-rules#21]
Is it possible to merge these two domain-list-community categories into their respective equivalents in this project?

category-ir -> ir
category-porn -> nsfw

**Comment by Chocolate4U** [comment]
Hi,
`category-ir` is completely merged in `ir` category and there is no need for it if using `ir` category.

`nsfw` is also more comprehensive than `category-porn` and there is no need for `category-porn` if using `nsfw` category.

Both `category-ir` and `category-porn` are only there for compatibility with [domain-list-community](https://github.com/v2fly/domain-list-community) rules, and are not needed if using their similar category from this repo.

**Comment by Chocolate4U** [comment]
Yes, that's why I didn't use their domain list for ads either. Many false-positives and poor maintenance.
Also, I recommend to use Cloudflare's family DNS (1.1.1.3) for blocking adult websites whenever possible. They tend to have a more complete database, and it will also reduce over head on sing-box.

**Comment by Chocolate4U** [comment]
A few things come to mind after seeing your config:
1. In DNS rule section, it's better to use `domain_suffix` to handle `.ir` TLD domains. Although I have included `.ir` in `geosite-ir`, using `domain_suffix` is a bit faster.
```
"rules": [
  {
    "domain_suffix": [
      ".ir"
    ],
    "rule_set": [
      "geosite-private",
      "geosite-ir"
    ],
    "server": "direct"
  },
  {
    "outbound": "any",
    "server": "direct"
  }
]
```

2. In DNS Server section, try to simplify it like this, and test if it works with those servers:
```
"servers": [
  {
    "address": "https://8.8.8.8/dns-query",
    "address_resolver": "direct",
    "tag": "remote",
    "detour": "default"
  },
  {
    "address": "local", //or 8.8.8.8
    "tag": "direct",
    "detour": "direct"
  }
]
```
In this configuration, proxy server's address gets resolved using local DNS server of the client. Which is specified in this rule:
```
{
  "outbound": "any",
  "server": "direct"
}
```
So no loop condition will occur. After that, all DNS queries (except domestic addresses, which get resolved using `local`) will go through `default` proxy server and get resolved using the specified DOH address. I use this configuration myself, and it works fine, but I'm not sure if it works for other servers with different settings. You can test it and verify.
Nice job on the configurations BTW.

**Comment by Chocolate4U** [comment]
Ok, in the case that you want to also resolve proxy addresses through DoH, I don't think there's an easy solution for this (or any solution FWIW). I suspect eventually every single DoH endpoint will get blocked inside the country, even the unknown ones.
There should be an entry point for proxy servers, either an IP address or a Domain address that's not blocked inside the country. In this case, the entry point is the DoH address, but I don't think trying to get the IP address from a blocked domain address by leveraging DoH, is sustainable in the long run, specially in the current level of censorship.
